server/src/main/java/org/apache/druid/segment/indexing/DataSchema.java

```diff
@@ -74,6 +74,9 @@ public DataSchema(
 
     Preconditions.checkArgument(!Strings.isNullOrEmpty(dataSource), "dataSource cannot be null or empty. Please provide a dataSource.");
     Preconditions.checkArgument(!dataSource.contains("/"), "dataSource cannot contain the '/' character.");
+    Preconditions.checkArgument(!dataSource.contains("\t"), "dataSource cannot contain the '\\t' character.");
```

Comment:
Suggest extracting these checks into a static method on DataSchema, could call it `validateDatasourceName`

Reply:
Should vertical tab (\v) and formfeed (\f) be disallowed as well?

Reply:
can use `\s` regex instead that covers `[ \t\n\x0B\f\r]`